packages/react/src/date-picker/DatePicker.test.tsx

```diff
@@ -170,8 +170,12 @@ describe('DatePicker', () => {
 		});
 		expect(container).toHTMLValidate({
 			extends: ['html-validate:recommended'],
-			// react 18s `useId` break this rule
-			rules: { 'valid-id': 'off' },
+			rules: {
+				// React 18's `useId` breaks this rule
+				'valid-id': 'off',
```

Comment:
That sounds like we should avoid React 18s `useId` function then instead of turning a "valid-id" rule off.

Reply:
Yeah, you could be onto something here. This particular option + comment were pre-existing, so I don't have any broader context. Having said that, and having just dealt with fixing the error for inline styles, I'm starting to think the issue lies with our config for this particular html validation library or the library itself and not with `useId` (though, maybe all of the above?). My reasoning being:

1. `style` attribute is not invalid html, so I don't understand why it was flagged.
2. The error message in the console does not guide me towards a solution or even where the error occurs. See screenshot below.

<img width="1341" alt="image" src="https://github.com/agriculturegovau/agds-next/assets/43688949/9bd78460-1f67-46a5-b38c-4f867ea74ee7">

I think this is worth a broader team discussion to see if we can remove the 63 instances of `'valid-id': 'off'`, perhaps by not using `useId` as you've suggested, and to see if we can tweak the `'html-validate:recommended'` config or find another library.

Reply:
@stowball ☝️ thoughts?

Reply:
Instead of not using `useId`, I'm interested in knowing what it is about the ids that make them invalid.

I would like to still use `useId` but maybe create a wrapper util for turning the ID into something valid.

eg. `useId()` might generate `1234` as the id. This is invalid because 1234 starts with a number so we could add "ID-" to the front to make it a valid ID.

We would need to be careful tough as I expect teams would be using IDs for automation testing. We don't want to break thousands of automation tests.

Reply:
>We would need to be careful tough as I expect teams would be using IDs for automation testing. We don't want to break thousands of automation tests.

Do you have examples of this type of test? We really should be encouraging folks to steer away from writing automated tests that lean on `id`s or `data-testid`s for getting elements in their tests. `id`s aren't a navigation source for users on a page. The very first part of [testing library's guiding principles](https://testing-library.com/docs/guiding-principles/) puts it beautifully:

>[The more your tests resemble the way your software is used, the more confidence they can give you.](https://twitter.com/kentcdodds/status/977018512689455106)

Reply:
In BARS we have a testing platform known as TOSCA. This platform runs a live version of the site in browser and is used to test the site the same way a user interacts with the site.

You need some way to select elements for TOSCA to click on and often IDs are used for this purpose.

Reply:
Yeah, I think we should look to update our config from `'html-validate:recommended'`.

`useId` returns an id like this: `':ro:'`.

While MDN states:

> Technically, the value for an id attribute may contain any character, except [whitespace](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace) characters. However, to avoid inadvertent errors, only [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) letters, digits, `'_'`, and `'-'` should be used, and the value for an id attribute should start with a letter.

If you put this in W3C's validator:

```
<!doctype html>
<html lang="en">
<head><title>foo</title></head>
<body>
<p id=":ro:">bar</p>
</body>
</html>
```

it will not have any errors. So, the `html-validate` package is just being overly prescriptive for no real reason.

Reply:
Have you tested that the IDs are picked up by a screen reader? If the id validation service is failing and we are assuming it is fine because of what the spec says, we need to test that it does indeed work. Browser/screen reader implementation does not always adhere perfectly with what the spec says.

Reply:
I think it would be safer to remove the `:` from the start and end of the ID to ensure it doesn't break any screen readers ability to read the IDs.

Reply:
>Have you tested that the IDs are picked up by a screen reader?

Yep, good question. Ids are picked up by the screen reader correctly and announcements are as expected 👍 

>I think it would be safer to remove the : from the start and end of the ID to ensure it doesn't break any screen readers ability to read the IDs.

Have you got evidence that this is a problem we need to solve? I've searched and can't find anything relating to `useId` causing any issues for screen readers.

Reply:
hmm... considering that [the documentation](https://react.dev/reference/react/useId#useid) says that `useId` is built specifically for the purpose of generating IDs intended for use in HTML for accessibility related attributes... it's probably been tested that it works for its intended purpose and is fine for us to use in this way.